source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1

```diff
@@ -0,0 +1,499 @@
+        Specifies the name of the SQL Server instance that clients use to communicate with the site system.
+
+    .PARAMETER Username
+        Specifies a domain user account that the management point uses to access site information.
+        If specifying an account the account must already exist in
+        Configuration Manager.  This can be achieved by:
+
+        $secure = ConvertTo-SecureString -String "Password" -AsPlainText -Force
+        New-CMAccount -Name 'contoso\test1' -Password $secure -SiteCode '<siteCode>'
+
+    .Notes
+        This must be ran on the Primary servers to install the ManagementPoint role.
+        The Primary server computer account must be in the local
+        administrators group to perform the install.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [String]
+        $SqlServerFqdn,
+
+        [Parameter()]
+        [String]
+        $DatabaseName,
+
+        [Parameter()]
+        [ValidateSet('Internet', 'Intranet', 'InternetAndIntranet')]
+        [String]
+        $ClientConnectionType,
+
+        [Parameter()]
+        [Boolean]
+        $EnableCloudGateway,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSsl,
+
+        [Parameter()]
+        [Boolean]
+        $GenerateAlert,
+
+        [Parameter()]
+        [Boolean]
+        $UseSiteDatabase,
+
+        [Parameter()]
+        [Boolean]
+        $UseComputerAccount,
+
+        [Parameter()]
+        [String]
+        $SqlServerInstanceName,
+
+        [Parameter()]
+        [String]
+        $Username,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+
+    try
+    {
+        if (($EnableCloudGateway -eq $true) -or ($state.EnableCloudGateway -eq $true))
+        {
+            if (($ClientConnectionType -eq 'Intranet') -or ([string]::IsNullOrEmpty($ClientConnectionType) -and
+                ([string]::IsNullOrEmpty($state.ClientConnectionType) -or $state.ClientConnectionType -eq 'Intranet')))
+            {
+                throw 'When CloudGateway is enabled, ClientConnectionType must not equal Intranet'
+            }
+
+            if (($PSBoundParameters.EnableSsl -eq $false) -or
+                ([string]::IsNullOrEmpty($PSBoundParameters.EnableSsl) -and
+                ([string]::IsNullOrEmpty($state.EnableSSL) -or $state.EnableSSL -eq $false)))
+            {
+                throw 'When CloudGateway is enabled SSL must also be enabled'
+            }
+        }
+
+        if ((($SqlServerFqdn) -and [string]::IsNullOrEmpty($DatabaseName)) -or
```

Comment:
This:($SqlServerFqdn)
Can be $SqlServerFqdn
But I understand if you keep it the same to show the seperation.
Same with $DatabaseName

Reply:
kept as was for separation